### PR TITLE
feat: add a built-in AI service

### LINKED ISSUE #474: 🚀 功能建议：添加一个单独的内置 AI 翻译服务
### 请先确认以下事项

- [X] 已仔细阅读了 [README](https://github.com/tisfeng/Easydict#readme)
- [X] 在 [issues](https://github.com/tisfeng/Easydict/issues) 页面搜索过（包括已关闭的 issue），未发现类似功能建议
- [X] Easydict 已升级到 [最新版本](https://github.com/tisfeng/Easydict/releases)
- [X] 我理解并认可上述内容，并理解项目维护者精力有限，**不遵循规则的 issue 可能会被无视或直接关闭**

### 功能描述

添加一个单独的内置 AI 翻译服务，移除对 OpenAI 服务的占用 [使用内置 APIKey](https://github.com/tisfeng/Easydict?tab=readme-ov-file#%E4%BD%BF%E7%94%A8%E5%86%85%E7%BD%AE-apikey) 

### 使用场景

某些 AI 服务会提供免费接口，例如 Google Gemini，但它需要代理，并且只对部分国家地区 IP 开放，对一些用户不友好。

借助 one-api，我们可以内置一个 AI 翻译，对外开放一些免费的模型服务，如 Gemini，Groq 等。

### 实现方案（可选）

_No response_

### 是否愿意提交 PR 实现该功能

- [X] 我愿意提交 PR

### PR DESCRIPTION
**Pull request by tisfeng**
close #474

image

### CODE REVIEW

**Review by Jerry23011 — DISMISSED** [review]

File: Easydict/App/Localizable.xcstrings [hunk 1]
```diff
@@ -619,6 +619,22 @@
         }
       }
     },
+    "built_in_ai" : {
```
Comment (phlpsong) [inline Easydict/App/Localizable.xcstrings:622]:
```suggestion
    "builtin_ai" : {
```

File: Easydict/Swift/Service/BuiltInAI/BuiltInAIService.swift [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    // MARK: Lifecycle
+
+    override init() {
+        super.init()
+    }
+
+    // MARK: Public
+
+    override public func name() -> String {
+        NSLocalizedString("built_in_ai", comment: "")
```
Comment (phlpsong) [inline Easydict/Swift/Service/BuiltInAI/BuiltInAIService.swift:22]:
```suggestion
        NSLocalizedString("builtin_ai", comment: "")
```

**Review by phlpsong — CHANGES_REQUESTED** [review]

**Review by phlpsong — APPROVED** [review]
LGTM

**Review by AkaShark — APPROVED** [review]

**Comment by AkaShark** [comment]
LGTM

**Comment by tisfeng** [comment]
I just added multiple available models 🙂

image

**Review by phlpsong — APPROVED** [review]

**Review by AkaShark — APPROVED** [review]

**Pull request merged**